client/graphics/texture-object.cpp

```diff
@@ -3,317 +3,300 @@
-		}
-		err = glGetError();
-		if (err) {
-			_log->trace("[Texture-Object] Error writing texture data on GPU");
-			return;
-		}
-		glGenerateMipmap(GL_TEXTURE_2D);
-		glBindTexture(GL_TEXTURE_2D, 0);
+	err = glGetError();
+	if (err) {
+		_log->trace("[Texture-Object] Error setting texture filters");
+		return;
+	}
+	if (usealpha) {
+		glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, width, height, 0, GL_RGBA, GL_UNSIGNED_BYTE, nullptr);
 	}
+	else {
+		glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, width, height, 0, GL_RGB, GL_UNSIGNED_BYTE, nullptr);
+	}
+	err = glGetError();
+	if (err) {
+		_log->trace("[Texture-Object] Error writing texture data on GPU");
+		return;
+	}
+	glGenerateMipmap(GL_TEXTURE_2D);
+	glBindTexture(GL_TEXTURE_2D, 0);
+}
 
-	void TextureObject::GenerateStencil(GLuint width, GLuint height) {
-		auto err = glGetError();
-		if (err) {
-			return;
-		}
-		if (!this->texture_id) {
-			glGenTextures(1, &this->texture_id);
-		}
-		glBindTexture(GL_TEXTURE_2D, this->texture_id);
-		err = glGetError();
-		if (err) {
-			return;
-		}
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
-		err = glGetError();
-		if (err) {
-			return;
-		}
+void TextureObject::GenerateStencil(GLuint width, GLuint height) {
+	auto err = glGetError();
+	if (err) {
+		return;
+	}
+	if (!this->texture_id) {
+		glGenTextures(1, &this->texture_id);
+	}
+	glBindTexture(GL_TEXTURE_2D, this->texture_id);
+	err = glGetError();
+	if (err) {
+		return;
+	}
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+	err = glGetError();
+	if (err) {
+		return;
+	}
 
-		glTexImage2D(GL_TEXTURE_2D, 0, GL_STENCIL_INDEX, width, height, 0, GL_STENCIL_INDEX, GL_UNSIGNED_BYTE, nullptr);
-		err = glGetError();
-		if (err) {
-			return;
-		}
+	glTexImage2D(GL_TEXTURE_2D, 0, GL_STENCIL_INDEX, width, height, 0, GL_STENCIL_INDEX, GL_UNSIGNED_BYTE, nullptr);
+	err = glGetError();
+	if (err) {
+		return;
+	}
 
-		glBindTexture(GL_TEXTURE_2D, 0);
+	glBindTexture(GL_TEXTURE_2D, 0);
+}
+
+void TextureObject::GenerateDepth(GLuint width, GLuint height, bool stencil) {
+	auto err = glGetError();
+	if (err) {
+		return;
+	}
+	if (!this->texture_id) {
+		glGenTextures(1, &this->texture_id);
 	}
+	glBindTexture(GL_TEXTURE_2D, this->texture_id);
+	err = glGetError();
+	if (err) {
+		return;
+	}
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+	if (this->compare) {
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_COMPARE_MODE, GL_COMPARE_REF_TO_TEXTURE);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_COMPARE_FUNC, GL_GEQUAL);
+	}
+	err = glGetError();
+	if (err) {
+		return;
+	}
+	if (stencil) {
```

Comment:
This could be cleaned up with a temp variable assigned to the texture type  GL_DEPTH_STENCIL or GL_DEPTH_COMPONENT

Reply:
I'm sure much of the GL code needs to be redone in places